src/awkward/_connect/_jax/jax_utils.py

```diff
@@ -76,9 +76,9 @@ def find_nparray_node(
 
         def recurse(outarray, recurse_where):
 
-            if isinstance(recurse_where, numbers.Integral) or isinstance(
-                recurse_where, str
-            ):
+            # TODO: capture makes this really confusing. A new parameter should
+            # be used instead if capture is really needed.
+            if isinstance(recurse_where, (numbers.Integral, str)):
```

Comment:
jax_utils.py is also on the chopping block. It's from the v1 JAX integration attempt, and @swishdiff will be doing this using an entirely different technique.

Reply:
Ahh, this was the main one that was driving me crazy. The capture makes this really hard to follow. And capture causes late binding, so it can be really tricky to correctly put this into a nicer form without knowing if the late binding is actually intended.

Reply:
Since this is code that will be replaced soon, the main thing is to learn from its example. We don't need to change it in this PR, but we'll just avoid this pattern in the future. (So I think I can resolve this one.)